src/components/ChatBotFooter/ChatBotFooter.tsx

```diff
@@ -46,18 +46,34 @@ const ChatBotFooter = ({
 	// handles options for bot
 	const { botOptions } = useBotOptions();
 
+	const renderFileAttachment = () => {
+		return (
+			<FileAttachmentButton inputRef={inputRef}
+				flow={flow} getCurrPath={getCurrPath} openChat={openChat}
+				getPrevPath={getPrevPath} handleActionInput={handleActionInput} injectMessage={injectMessage}
+				streamMessage={streamMessage}
+			/>
+		)
+	}
+
+	const renderEmojiPicker = () => {
+		return (<EmojiPicker inputRef={inputRef} textAreaDisabled={textAreaDisabled}/>)
+	}
+	
 	return (
 		<div style={botOptions.footerStyle} className="rcb-chat-footer-container">
 			<div className="rcb-chat-footer">
-				{!botOptions.fileAttachment?.disabled &&
-					<FileAttachmentButton inputRef={inputRef} flow={flow} getCurrPath={getCurrPath} openChat={openChat}
-						getPrevPath={getPrevPath} handleActionInput={handleActionInput} injectMessage={injectMessage}
-						streamMessage={streamMessage}
-					/>
-				}
-				{!botOptions.emoji?.disabled &&
-					<EmojiPicker inputRef={inputRef} textAreaDisabled={textAreaDisabled}/>
-				}
+				{botOptions.footer?.buttons?.map((element) => {
+					if(element === "file-attachment" && !botOptions.fileAttachment?.disabled){
```

Comment:
The styling is inconsistent here (whitespace is missing), check for the rest of the changes as well.